maat-court-data-api/src/main/java/gov/uk/courtdata/iojappeal/impl/IOJAppealImpl.java

```diff
@@ -25,7 +26,12 @@ public IOJAppealEntity findByRepId(int repId) {
     }
 
     public IOJAppealEntity create(IOJAppealDTO iojAppealDTO) {
-        var iojAppealEntity = iojAppealMapper.toIOJIojAppealEntity(iojAppealDTO);
+        var iojAppealEntity = iojAppealMapper.toIojAppealEntity(iojAppealDTO);
+        return iojAppealRepository.save(iojAppealEntity);
+    }
+
+    public IOJAppealEntity create(ApiCreateIojAppealRequest apiCreateIojAppealRequest) {
```

Comment:
We’ve got a similar issue here, made more obvious by the existing overloads. The persistence layer shouldn’t be aware of API-level request shapes (DTOs or V2 request models). It should expose a single create(`IOJAppealEntity`) method, with all mapping handled in the V1/V2 service classes. Keeping the persistence layer entity-only avoids version awareness and keeps the responsibilities clean.